pkg/controller/pxc/upgrade.go

```diff
@@ -118,6 +118,31 @@ func (r *ReconcilePerconaXtraDBCluster) updatePod(sfs api.StatefulApp, podSpec *
 	return r.client.Update(context.TODO(), currentSet)
 }
 
+func (r *ReconcilePerconaXtraDBCluster) updateService(svc *corev1.Service, podSpec *api.PodSpec) error {
+	currentService := &corev1.Service{}
+	err := r.client.Get(context.TODO(), types.NamespacedName{Name: svc.Name, Namespace: svc.Namespace}, currentService)
+	if err != nil {
+		return fmt.Errorf("failed to get sate: %v", err)
+	}
+
+	if podSpec.ServiceType != nil && currentService.Spec.ClusterIP != "None" {
+		switch *podSpec.ServiceType {
+		case corev1.ServiceTypeClusterIP:
+			currentService.Spec.Ports = []corev1.ServicePort{
+				{
+					Port: 3306,
+					Name: "mysql",
+				},
```

Comment:
Is it enough? Since here we have, for example, https://github.com/percona/percona-xtradb-cluster-operator/blob/master/pkg/pxc/service.go#L150-L160

Reply:
In fact, update logic works only if ClusterIP='xxx.xxx.xxx.xxx' and that is true only for ProxySQL

Reply:
@cap1984 should we set Protocol and TargetPort also in lines 132-134?